planning/behavior_path_planner/src/scene_module/lane_change/normal.cpp

```diff
@@ -572,10 +581,11 @@ bool NormalLaneChange::getLaneChangePaths(
       (prepare_velocity - current_velocity) / prepare_duration;
 
     // get path on original lanes
-    const double prepare_length = std::max(
+    const double expected_prepare_length =
       current_velocity * prepare_duration +
-        0.5 * longitudinal_acc_on_prepare * std::pow(prepare_duration, 2),
-      minimum_prepare_length);
+      0.5 * longitudinal_acc_on_prepare * std::pow(prepare_duration, 2);
+    const auto prepare_length =
+      !is_near_end_of_road || (getEgoVelocity() >= 1.0) ? expected_prepare_length : 0.0;
```

Comment:
I think you need to make the `prepare_duration = 0.0' when it satisfies the condition. Since `prepare_duration` is used in the lane change path generation and safety check, you should change this variable here.

Reply:
My suggestion is like 
```C++
const bool disable_prepare_segment = is_near_end_of_road && (getEgoVelocity() < 1.0);
const double prepare_duration = disable_prepare_segment ? 0.0 : common_parameter.prepare_duration;

const double prepare_length  = current_velocity * prepare_duration + 0.5 * longitudinal_acc_on_prepare * std::pow(prepare_duration, 2);
```

Reply:
fixed in [c4c1405](https://github.com/autowarefoundation/autoware.universe/pull/4242/commits/c4c140511a6a695224eef58b828a45c232dcfab4)